Tools/CISupport/ews-build/events.py

```diff
@@ -164,13 +222,46 @@ def buildFinished(self, key, build):
             "steps": build.get('steps'),
         }
 
-        self.sendData(data)
+        self.sendDataToEWS(data)
+
+    @defer.inlineCallbacks
+    def stepStartedGitHub(self, build, state_string):
+        sha = self.extractProperty(build, 'revision')
+        repository = self.extractProperty(build, 'repository')
+        if not sha or not repository:
+            return
+
+        builder = yield self.master.db.builders.getBuilder(build.get('builderid'))
+
+        self.sendDataToGitHub(
+            repository, sha, dict(
+                owner=(self.extractProperty(build, 'owners') or [None])[0],
+                repo=(self.extractProperty(build, 'github.head.repo.full_name') or '').split('/')[-1],
+                sha=sha,
+                target_url='{}#/builders/{}/builds/{}'.format(self.master.config.buildbotURL, build.get('builderid'), build.get('number')),
+                state={
+                    FAILURE: 'failure',
+                    EXCEPTION: 'error',
+                }.get(build.get('results'), 'pending'),
```

Comment:
Better to have explicit entry in this dictionary indicating how we are handling other major states like success, warning etc, even if we are mapping them to 'pending'.